dataprofiler/profilers/profiler_options.py

```diff
@@ -376,6 +376,45 @@ def _validate_helper(self, variable_path='IntOptions'):
         return super()._validate_helper(variable_path) 
 
 
+class PrecisionOptions(BooleanOption):
+    
+    def __init__(self, is_enabled=True, sample_ratio=None):
+        """
+        Options for precision
+
+        :ivar is_enabled: boolean option to enable/disable the column.
+        :vartype is_enabled: bool
+        :ivar sample_ratio: float option to determine ratio of valid
+                            float samples in determining percision.
+                            This ratio will override any defaults.
+        :vartype sample_ratio: float
+        """
+        self.sample_ratio = sample_ratio
+        super().__init__(is_enabled=is_enabled)
+
+    def _validate_helper(self, variable_path='PrecisionOptions'):
+        """
+        Validates the options do not conflict and cause errors.
+        
+        :param variable_path: current path to variable set.
+        :type variable_path: str
+        :return: list of errors (if raise_error is false)
+        :rtype: List of strings
+        """
+        errors = super()._validate_helper(variable_path=variable_path)    
+        if self.sample_ratio is not None:
+            if not isinstance(self.sample_ratio, float) \
+               and not isinstance(self.sample_ratio, int):
+                errors.append("{}.sample_ratio must be a float and cannot be a string."
```

Comment:
I don't think there is any point specifying that it "cannot be a string", just saying it must be a float seems sufficient